posydon/interpolation/profile_interpolation.py

```diff
@@ -192,60 +208,86 @@ def load_profiles(self,filename,valid_split=0.2):
-                                self.valid_initial, 
-                                self.valid_profiles[:,self.names.index("x_mass_fraction_H")], 
-                                self.valid_profiles[:,self.names.index("y_mass_fraction_He")], 
-                                self.valid_scalars["star_state"], 
-                                IF_interpolator,
-                                comp_bounds_epochs,comp_bounds_patience,hms_s2)
-
-        # instantiate and train density profile model
-        self.dens = Density(self.initial,
-                            self.profiles[:,self.names.index("logRho")],
-                            self.valid_initial,
-                            self.valid_profiles[:,self.names.index("logRho")],
-                            IF_interpolator,hms_s2=hms_s2)
-        self.dens.train(prof_epochs=density_epochs,prof_patience=density_patience)
+        if comp==True:
+            # instantiate and train composition (H and He mass fraction) profiles model
+            self.comp = Composition(self.initial, 
+                                    self.profiles[:,self.names.index("x_mass_fraction_H")], 
+                                    self.profiles[:,self.names.index("y_mass_fraction_He")], 
+                                    self.scalars["star_state"], 
+                                    self.valid_initial, 
+                                    self.valid_profiles[:,self.names.index("x_mass_fraction_H")], 
+                                    self.valid_profiles[:,self.names.index("y_mass_fraction_He")], 
+                                    self.valid_scalars["star_state"], 
+                                    IF_interpolator,
+                                    comp_bounds_epochs,comp_bounds_patience,hms_s2)
+        if density==True:
+            # instantiate and train density profile model
+            self.dens = Density(self.initial,
+                                self.profiles[:,self.names.index("logRho")],
+                                self.scalars["MT_class"],
+                                self.valid_initial,
+                                self.valid_profiles[:,self.names.index("logRho")],
+                                self.valid_scalars["MT_class"],
+                                IF_interpolator,hms_s2=hms_s2,
+                                depth=depth, width=width,
+                                depthn=depthn, widthn=widthn)
+            self.dens.train(prof_epochs=density_epochs,prof_patience=density_patience,lr=lr)
         
         if loss_history==True:
-            return self.comp.loss_history, self.dens.loss_history
-
-    def predict(self,inputs):
+            if comp==True and density==True:
+                return self.dens.loss_history, self.comp.loss_history
+            elif comp!=True and density==True:
+                return self.dens.loss_history
+            elif comp==True and density!=True:
+                return self.comp.loss_history
```

Comment:
Please reflect the three return options in the doc-string. I think it is important to specify well, which order the two histories will have in case both are returned. (it looks that the order switched)
Additionally, please consider to add a general else case with a return to not let the function ever exit without a return.

Reply:
done!